xarray/core/dataarray.py

```diff
@@ -234,7 +258,7 @@ class DataArray(AbstractArray, DataWithCoords):
 
     def __init__(
         self,
-        data: Any,
+        data: Any = dtypes.NA,
```

Comment:
This means that an error won't be raised when calling `DataArray` with no parameters. Is there a better way to do this? 
e.g.
```python
>>> xr.DataArray()

<xarray.DataArray ()>
array(nan)
```
is returned instead of
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-35-657adafd8cc6> in <module>
----> 1 xr.DataArray()

TypeError: __init__() missing 1 required positional argument: 'data'
```

Reply:
I'm not sure that's a bad thing. 

We could easily make it raise, by checking that coords are passed if `data = NA`. But not sure that's necessary?

Reply:
> I'm not sure that's a bad thing.

Agreed, I think the current behavior here makes sense.